pkg/codegen/docs_integration_test.go

```diff
@@ -318,6 +291,279 @@ func TestGetLanguageTypeString(t *testing.T) {
+	t.Parallel()
+
+	schemaSpec := schema.PackageSpec{
+		Name: "example",
+		Resources: map[string]schema.ResourceSpec{
+			"example:index:Foo": {
+				IsComponent: true,
+				Methods: map[string]string{
+					"getKubeconfig": "example:index:Foo/getKubeconfig",
+				},
+			},
+		},
+		Functions: map[string]schema.FunctionSpec{
+			"example:index:Foo/getKubeconfig": {
+				Inputs: &schema.ObjectTypeSpec{
+					Properties: map[string]schema.PropertySpec{
+						"__self__": {
+							TypeSpec: schema.TypeSpec{
+								Ref: "#/resources/example:index:Foo",
+							},
+						},
+						"profileName": {
+							TypeSpec: schema.TypeSpec{
+								Type: "string",
+							},
+						},
+						"roleArn": {
+							TypeSpec: schema.TypeSpec{
+								Type: "string",
+							},
+						},
+					},
+					Required: []string{"__self__"},
+				},
+				Outputs: &schema.ObjectTypeSpec{
+					Properties: map[string]schema.PropertySpec{
+						"kubeconfig": {
+							TypeSpec: schema.TypeSpec{
+								Type: "string",
+							},
+						},
+					},
+					Required: []string{"kubeconfig"},
+				},
+			},
+		},
+	}
+
+	pkg, err := schema.ImportSpec(schemaSpec, nil, schema.ValidationOptions{
+		AllowDanglingReferences: true,
+	})
+	require.NoError(t, err)
+
+	expected := map[language]string{
+		golang: "FooGetKubeconfigResultOutput",
+		nodejs: "Foo.GetKubeconfigResult",
+		python: "Foo.Get_kubeconfigResult",
+		dotnet: "Foo.GetKubeconfigResult",
+	}
+
+	// Code generation is not safe to parallelize since import binding mutates the
+	// [schema.Package].
+	for lang, expected := range expected {
+		testDocsGenHelper(t, lang, pkg.Reference(), func(t *testing.T, helper codegen.DocLanguageHelper) {
+			actual := helper.GetMethodResultName(pkg.Reference(), "",
+				mustToken(t, pkg.Reference().Resources().Get, "example:index:Foo"),
+				mustToken(t, pkg.Reference().Resources().Get, "example:index:Foo").Methods[0],
+			)
+			assert.Equal(t, expected, actual)
+		})
+	}
+}
+
+func testDocsGenHelper(
+	t *testing.T, language language, schema schema.PackageReference,
+	f func(*testing.T, codegen.DocLanguageHelper),
+) {
+	var name string
+	var helper func() codegen.DocLanguageHelper
+	switch language {
+	case nodejs:
+		helper = mkHelper[nodejs_codegen.DocLanguageHelper]
+		name = "nodejs"
+	case python:
+		helper = mkHelper[python_codegen.DocLanguageHelper]
+		name = "python"
+	case golang:
+		helper = func() codegen.DocLanguageHelper {
+			h := golang_codegen.DocLanguageHelper{}
+			var info golang_codegen.GoPackageInfo
+			if i, err := schema.Language("go"); err == nil && i != nil {
+				info = i.(golang_codegen.GoPackageInfo)
+			}
+			h.GeneratePackagesMap(schema, "test", info)
+			return h
+		}
+		name = "go"
+	case dotnet:
+		helper = mkHelper[dotnet_codegen.DocLanguageHelper]
+		name = "dotnet"
+	default:
+		assert.Fail(t, "Unknown language %T", language)
+	}
+
+	t.Run(name, func(t *testing.T) { //nolint:paralleltest // golangci-lint v2 upgrade
```

Comment:
This is no longer part of any upgrade, can we just fix it?  Can we just add `t.Parallel()` here?

Reply:
I'll do that in a follow-up PR. I'd like this to go into the registry ASAP since it's now necessary to accept new TF based providers.